src/main/scala/no/ndla/articleapi/service/search/SearchService.scala

```diff
@@ -10,46 +10,87 @@
 import java.lang.Math.max
-import com.google.gson.JsonObject
+
+import com.google.gson.{JsonObject, JsonParser}
 import com.typesafe.scalalogging.LazyLogging
 import io.searchbox.core.{Count, SearchResult}
-import no.ndla.articleapi.ArticleApiProperties.{MaxPageSize, DefaultPageSize}
+import no.ndla.articleapi.ArticleApiProperties.{DefaultPageSize, MaxPageSize}
 import no.ndla.articleapi.integration.ElasticClient
 import no.ndla.articleapi.model.domain
 import no.ndla.articleapi.model.domain._
-import org.elasticsearch.search.sort.{FieldSortBuilder, SortBuilders, SortOrder}
+import no.ndla.articleapi.service.ConverterService
+import org.elasticsearch.script.Script
+import org.elasticsearch.search.sort.ScriptSortBuilder.ScriptSortType
+import org.elasticsearch.search.sort._
+import org.json4s._
+import org.json4s.native.JsonMethods._
 
 trait SearchService {
-  this: ElasticClient with LazyLogging =>
+  this: ElasticClient with ConverterService with LazyLogging =>
 
   trait SearchService[T] {
     val searchIndex: String
 
     def hitToApiModel(hit: JsonObject, language: String): T
 
-    def getHits(response: SearchResult, language: String): Seq[T] = {
-      var resultList = Seq[T]()
+    def getHits(response: SearchResult, language: String, hitToApi:(JsonObject, String) => T): Seq[T] = {
       response.getTotal match {
         case count: Integer if count > 0 =>
-          val resultArray = response.getJsonObject.get("hits").asInstanceOf[JsonObject].get("hits").getAsJsonArray
-          val iterator = resultArray.iterator()
-          while (iterator.hasNext) {
-            resultList = resultList :+ hitToApiModel(iterator.next().asInstanceOf[JsonObject].get("_source").asInstanceOf[JsonObject], language)
-          }
-          resultList
-        case _ => Seq.empty
+          val resultArray = (parse(response.getJsonString) \ "hits" \ "hits").asInstanceOf[JArray].arr
+
+          resultArray.map(result => {
+            val matchedLanguage = language match {
+              case Language.AllLanguages | "*" => converterService.getLanguageFromHit(result).getOrElse(language)
+              case _ => language
+            }
+
+            val hitString = compact(render(result \ "_source"))
+            val jsonSource = new JsonParser().parse(hitString).getAsJsonObject
+            hitToApi(jsonSource, matchedLanguage)
+          })
+        case _ => Seq()
       }
     }
 
-    def getSortDefinition(sort: Sort.Value, language: String): FieldSortBuilder = {
+    def getSortDefinition(sort: Sort.Value, language: String) = {
       val sortLanguage = language match {
         case domain.Language.NoLanguage => domain.Language.DefaultLanguage
         case _ => language
       }
 
+      // Elasticsearch 'Painless' script for sorting by title if searching for all languages
+      val supportedLanguages = Language.languageAnalyzers.map(la => la.lang).mkString("'", "', '", "'")
+      val titleSortScript =
+        s"""
+           |int idx = -1;
```

Comment:
Fantastisk at man må skrive et eget script for å sortere på tittel som har flere språkfelt.. :p